File: src/main/java/icu/samnyan/aqua/sega/maimai2/handler/GetUserExtendHandler.java
package icu.samnyan.aqua.sega.maimai2.handler;

import com.fasterxml.jackson.core.JsonProcessingException;
import icu.samnyan.aqua.sega.maimai2.model.Mai2UserExtendRepo;
import icu.samnyan.aqua.sega.general.BaseHandler;
import icu.samnyan.aqua.sega.maimai2.model.userdata.Mai2UserExtend;
import icu.samnyan.aqua.sega.util.jackson.BasicMapper;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * @author samnyan (privateamusement@protonmail.com)
 */
@Component("Maimai2GetUserExtendHandler")
public class GetUserExtendHandler implements BaseHandler {

    private static final Logger logger = LoggerFactory.getLogger(GetUserExtendHandler.class);

    private final BasicMapper mapper;

    private final Mai2UserExtendRepo userExtendRepository;

    public GetUserExtendHandler(BasicMapper mapper, Mai2UserExtendRepo userExtendRepository) {
        this.mapper = mapper;
        this.userExtendRepository = userExtendRepository;
    }

    @Override
    public String handle(Map<String, Object> request) throws JsonProcessingException {
        long userId = ((Number) request.get("userId")).longValue();

        Mai2UserExtend userExtend = userExtendRepository.findSingleByUser_Card_ExtId(userId).orElseThrow();

        Map<String, Object> resultMap = new LinkedHashMap<>();
        resultMap.put("userId", userId);
        resultMap.put("userExtend", userExtend);

        String json = mapper.write(resultMap);
        logger.info("Response: " + json);
        return json;
    }
}

File: src/main/java/icu/samnyan/aqua/sega/maimai2/handler/GetUserExtendHandler.kt
package icu.samnyan.aqua.sega.maimai2.handler

import ext.invoke
import ext.minus
import ext.toJson
import icu.samnyan.aqua.sega.general.BaseHandler
import icu.samnyan.aqua.sega.maimai2.model.Mai2Repos
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

/**
 * @author samnyan (privateamusement@protonmail.com)
 */
@Component("Maimai2GetUserExtendHandler")
class GetUserExtendHandler(val repos: Mai2Repos) : BaseHandler {
    override fun handle(request: Map<String, Any>): Any {
        val userId = (request["userId"] as Number).toLong()

        val userExtend = repos.userExtend.findSingleByUser_Card_ExtId(userId)() ?: (404 - "User Extend Not Found")
        val resultMap = mapOf(
            "userId" to userId,
            "userExtend" to userExtend
        )

        val json = resultMap.toJson()
        logger.info("Response: $json")
        return json
    }

    companion object {
        val logger = LoggerFactory.getLogger(GetUserExtendHandler::class.java)
    }
}
